### ISSUE TITLE
Issue when using `rspec` with `--bisect` option

### ISSUE THREAD

**Issue by pucinsk** [tristandunn/pusher-fake#66]
**Description**
When I run `rspec` with `--bisect` option enabled then I get an error regarding `$stdout` usage.

**Error**
```ruby
bundle exec rspec --bisect spec/any_spec.rb
...
# pusher-fake/lib/pusher-fake/configuration.rb:53:in `reset!'
NoMethodError:
  undefined method `to_io' for #<StringIO:0x00007fa3e189ecc0>
  Did you mean?  to_s
                 to_a
                 to_h
```

This error occurs because `RSpec::Core::Bisect::ForkRunner` overrides `$stdout` with `StringIO` [here](https://github.com/rspec/rspec-core/blob/main/lib/rspec/core/bisect/fork_runner.rb#L84).

Any ideas how it could be fixed? 🙏 

Versions:
```
rspec 3.10
pusher-fake 3.0.0
```

Thank you for your time

**Comment by tristandunn** [comment]
Good catch and thanks for letting me know. The fix might be as easy as a condition to verify the object responds to `to_io`:

```diff
-      self.logger   = $stdout.to_io
+      self.logger   = $stdout.respond_to?(:to_io) ? $stdout.to_io : $stdout
```

I'll dig into it more tonight and hopefully get a fix out.

Related, I found an old Ruby bug, [#5479](https://bugs.ruby-lang.org/issues/5479), that discusses adding `StringIO#to_io` but looks like it was rejected 10 years ago.

**Comment by tristandunn** [comment]
@pucinsk Just released `3.0.1` to [GitHub](https://github.com/tristandunn/pusher-fake/packages/318630) and [RubyGems](https://rubygems.org/gems/pusher-fake/versions/3.0.1). Let me know if you have any more issues.

**Comment by pucinsk** [comment]
Thank you very much @tristandunn !
`rspec --bisect` works. All good!
